packages/opentelemetry-metrics/src/export/types.ts

```diff
@@ -0,0 +1,370 @@
+ * population. The bucket boundaries for that histogram are described by
+ * BucketOptions.
+ *
+ * If bucket_options has no type, then there is no histogram associated with
+ * the Distribution.
+ */
+export interface BucketOptions {
+  /** Bucket with explicit bounds. */
+  readonly explicit: Explicit;
+}
+
+/**
+ * Properties of an Explicit.
+ * Specifies a set of buckets with arbitrary upper-bounds.
+ * This defines size(bounds) + 1 (= N) buckets. The boundaries for bucket
+ * index i are:
+ *
+ * [0, bucket_bounds[i]) for i == 0
+ * [bucket_bounds[i-1], bucket_bounds[i]) for 0 < i < N-1
+ * [bucket_bounds[i-1], +infinity) for i == N-1
+ */
+export interface Explicit {
+  /** The values must be strictly increasing and > 0. */
+  readonly bounds: number[];
+  // TODO: If OpenMetrics decides to support (a, b] intervals we should add
+  // support for these by defining a boolean value here which decides what
+  // type of intervals to use.
+}
+
+/** Properties of a Bucket. */
+export interface Bucket {
+  /**
+   * The number of values in each bucket of the histogram, as described in
+   * bucket_bounds.
+   */
+  readonly count: number;
+  /**
+   * If the distribution does not have a histogram, then omit this field.
+   */
+  readonly exemplar?: Exemplar;
+}
+
+/**
+ * Exemplars are example points that may be used to annotate aggregated
+ * Distribution values. They are metadata that gives information about a
+ * particular value added to a Distribution bucket.
+ */
+export interface Exemplar {
+  /**
+   * Value of the exemplar point. It determines which bucket the exemplar
+   * belongs to.
+   */
+  readonly value: number;
+  /** The observation (sampling) time of the above value. */
+  readonly timestamp: Timestamp;
+  /** Contextual information about the example value. */
+  readonly attachments: { [key: string]: string };
+}
+
+/**
+ * The values in this message can be reset at arbitrary unknown times, with
+ * the requirement that all of them are reset at the same time.
+ */
+export interface Snapshot {
+  /**
+   * The number of values in the snapshot. Optional since some systems don't
+   * expose this.
+   */
+  readonly count: number;
+  /**
+   * The sum of values in the snapshot. Optional since some systems don't
+   * expose this. If count is zero then this field must be zero or not set
+   * (if not supported).
+   */
+  readonly sum: number;
+  /**
+   * A list of values at different percentiles of the distribution calculated
+   * from the current snapshot. The percentiles must be strictly increasing.
+   */
+  readonly percentileValues: ValueAtPercentile[];
+}
+
+/**
+ * Represents the value at a given percentile of a distribution.
+ */
+export interface ValueAtPercentile {
+  /** The percentile of a distribution. Must be in the interval (0.0, 100.0]. */
+  readonly percentile: number;
+  /** The value at the given percentile of a distribution. */
+  readonly value: number;
+}
+
+export interface Timestamp {
```

Comment:
Any reason not to use the HrTime interface we have for spans https://github.com/open-telemetry/opentelemetry-js/blob/7fff5383abf343a49e1f04e39911d65ef0a0a673/packages/opentelemetry-types/src/common/Time.ts#L18?

Or maybe refactor spans to use `Timestamp`? Feels like ideally trace and metrics would use the same time model

Reply:
Absolutely - I fix that. This is a leftover from OC.